source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,373 @@
+                             "This can for example be used to exclude the composition representing a sticky air, which "
+                             "is as cold as the surface, but you do not want to be refined."
+                            );
+          prm.declare_entry ("List of isotherms", "",
+                             Patterns::List(Patterns::List(Patterns::Anything(),0,100000000,","),0,100000000,";"),
+                             "A list of isotherms separated by semi-colins (;). Each isotherm entry consists of "
+                             "four values separted by a comma. The first two values indicate the minimum and maximum "
+                             "refinement levels respecitvely. The can be an integer, but also the words min or max "
+                             "optionally followed by a plus or minus a number. For example 'min+2' or 'max-1'. The "
+                             "last two values are the mimum and maximum temperature respectively between which the "
+                             "mimumun and maximum refinement are enforced."
+                            );
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      std::vector<std::pair<std::string,std::string> > isotherms_raw;
+      prm.enter_subsection("Mesh refinement");
+      {
+        minimum_refinement_level = Utilities::string_to_int(prm.get ("Minimum refinement level"));
+        maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isotherms");
+        {
+          exclude_composition = Utilities::string_to_int(prm.get("Exclude composition"));
+          std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+          // process the List of isotherms to get all data out of the structure
+          isotherms_raw.resize(isotherms_outer_loop.size());
+          isotherms.resize(isotherms_outer_loop.size());
+          // loop through all given isotherms
+          for ( unsigned int isotherm_line = 0; isotherm_line != isotherms_raw.size(); ++isotherm_line )
+            {
+              std::vector<std::string> isotherms_inner_loop = Utilities::split_string_list(isotherms_outer_loop[isotherm_line]);
+
+              // The minimum and maximum refinemnt levels respecively
+              isotherms_raw[isotherm_line].first  = isotherms_inner_loop[0];
+              isotherms_raw[isotherm_line].second = isotherms_inner_loop[1];
+              // The temperatures for the min and max isotherms respectively
+              isotherms[isotherm_line].first  = Utilities::string_to_double(isotherms_inner_loop[2]);
+              isotherms[isotherm_line].second = Utilities::string_to_double(isotherms_inner_loop[3]);
```

Comment:
what happens if you enter something in the input file that is not actually a number?